src/storage-types/src/sources.rs

```diff
@@ -639,6 +639,12 @@ pub trait SourceConnection: Debug + Clone + PartialEq + crate::AlterCompatible {
     /// The name of the resource in the external system (e.g kafka topic) if any
     fn upstream_name(&self) -> Option<&str>;
 
+    /// The schema of this connection's key rows.
+    fn key_desc(&self) -> RelationDesc;
```

Comment:
The comment from the PR description would be helpful here:

>  This is mostly setting the stage for the subsequent PRs that will attempt to compute and typecheck subsequent stages of the pipeline using the types of the earlier stages of the pipeline.

Was curious to me that this wasn't used anywhere and that it presents the schema independent of the encoding. I think this makes sense in context of the PR description, but this context isn't provided in the commit message, either.

Reply:
Good point, will add